teps/0056-pipelines-in-pipelines.md

```diff
@@ -352,6 +363,451 @@ main-`PipelineRun` to be able to fetch and access the sub-`PipelineRun`'s full s
 6. Users should be able to propagate actions from the main-`Pipeline` to the sub-`Pipeline`, such as deletion and
 cancellation.
 
+## Proposal
+
+This proposal focuses on enabling specification and execution of a `Pipeline` in a `Pipeline`. This section focuses 
+on the API change, see the [design](#design) section below for further details.
+
+### Specification
+
+To support defining `Pipelines` in `Pipelines` at authoring time, we propose adding `PipelineRef` and `PipelineSpec` 
+fields to [`PipelineTask`][pipelinetask] alongside `TaskRef` and `TaskSpec`. The `PipelineSpec` field will take a new 
+type, `EmbeddedPipeline`, similar to the [`EmbeddedTask`][embeddedtask] type, in order to support specifying metadata
```

Comment:
Do we need to support `EmbeddedPipeline`? What about the `pipelineSpec` without using it with pipelines in pipeline? What kind of use case are we trying to solve with this? How are we defining the precedence order for the `metadata`?

Reply:
Maybe we don't -- looks like we got too much into the implementation details so updating the TEP to focus on the API